draft-rdb-ohai-feedback-to-proxy-06.txt

```diff
@@ -0,0 +1,784 @@
+
+Internet-Draft          Oblivious Relay Feedback            October 2022
+
+
+3.  Providing RateLimit Information to an Oblivious Proxy
+
+   An Oblivious resource that uses RateLimit fields [RATELIMIT] to
+   return service limit information MAY add the "ohttp-target" quota
+   policy parameter defined in Section 4 to signal to the receiver that
+   the associated quota policy is intended for an Oblivious relay.  For
+   example, when an Oblivious target identifies a high frequency or high
+   volume anomalies in the HTTP requests it would include the "ohttp-
+   target" parameter.
+
+   The term "Oblivious Relay Feedback" denotes both the mechanism
+   described in this specification and the complete set of RateLimit
+   fields together with the "ohttp-target" parameter.
+
+   To know whether the RateLimit fields provides Oblivious Relay
+   Feedback (see Section 3.1), an Oblivious relay MUST:
+
+   1.  Identify the quota policy associated to the expiring limit.
+
+   2.  Check whether the "ohttp-target" parameter is present and its
+       syntax is correct.
+
+   In the example shown in Figure 1, the expiring limit value is "100",
+   so the associated quota policy is the second one.  This quota policy
+   includes the "ohttp-target" parameter: this indicates that the
+   RateLimit fields are intended for an Oblivious relay.
+
+
+      RateLimit-Limit: 100
+      RateLimit-Policy: 10;w=1, 100;w=60;ohttp-target=1
+      RateLimit-Remaining: 8
+      RateLimit-Reset: 15
+
+             Figure 1: An Example of Oblivious Proxy Feedback.
+
+4.  The ohttp-target Quota Policy Parameter
+
+4.1.  ohttp-target Parameter
+
+   The following quota policy parameter is defined for the RateLimit-
+   Policy field [RATELIMIT]:
+
+   ohttp-target:  Indicates that the associated quota policy provides
+      Oblivious Relay Feedback.  This parameter is OPTIONAL.
+
+   The "ohttp-target" parameter has the following syntax:
+
+   ohttp-target = sf-integer
+
+
+
+Reddy, et al.             Expires 8 April 2023                  [Page 4]
+
+Internet-Draft          Oblivious Relay Feedback            October 2022
+
+
+   Its value MUST be an Integer (Section 3.3.1 of [STRUCTURED-FIELDS])
+   and indicates whether the quota policy is applicable to all the
+   clients that are serviced by the Oblivious relay or applicable only
+   to a specific client.  The "ohttp-target" parameter MUST have one of
+   the following values:
+
+   1:  Indicates that RateLimit fields are applicable to all the clients
+      that are serviced by the same Oblivious relay.
+
+   2:  Indicates that RateLimit fields are applicable only to the
+      offending client.  For example, this value is used if the client
+      is attacking the server (e.g., the client is using an abnormal
+      header that matches an attack pattern).  The Oblivious relay MUST
+      NOT enforce the requested rate limit unconditionally otherwise the
```

Comment:
Absent a characterization of what is meant by "unconditionally", it will be for an implementation to follow this MUST NOT

Reply:
I did not get the comment.

Reply:
i.e., What means "unconditionally" here?

Reply:
I meant the relay cannot enforce the rate-limit rule blindly, it needs checks on its own to ensure it is not adversely impacting the privacy of clients (using the mechanisms discussed in Section 5).

Reply:
OK. It is worth to elaborate this as part of the proposed text.

Reply:
It is elaborated by the text in the next line:
The Oblivious relay MUST NOT enforce the requested rate limit unconditionally otherwise the target can take advantage of the differential treatment applied by the relay to deanonymize the client.  The mechanisms that relay can use to rate-limit the traffic from the offending client but at the same time prevent the target to correlate requests from clients are discussed in Section 5.

Reply:
How about, more succinctly:

`Naïve application of the rate limit can allow the target to de-anonymize the client so the Oblivious relay MUST follow the guidelines discussed in Section 5.`